src/content/docs/apis/nerdgraph/examples/browser-monitoring-config-nerdgraph.mdx

```diff
@@ -302,4 +302,119 @@ Here's the response:
+- **Efficiency**: Reduce deploy time and mitigate deployment difficulties.
+
+Here is an overview of how to use the Browser Agent Version Pinning API:
+
+1. Check agent type and which version is currently pinned, if any
+1. Reference the available versions of the agent
+1. Select and pin the desired version
+1. Redeploy or restart your service to use the pinned version of the agent
+
+### Step 1. Retrieve the Current Pinned Version
+
+
+Use this NerdGraph query to determine which agent type your browser application uses as well as if a specific version is already pinned.
+
+
+```graphql
+query FetchBrowserMonitoringAgentSettings {
+  actor {
+    entity(guid: "YOUR_BROWSER_APP_GUID") {
+      ... on BrowserApplicationEntity {
+        guid
+        name
+        browserSettings {
+          browserMonitoring {
+            loader
+            pinnedVersion
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+An example response to the above query:
+
+```json
+{
+  "data": {
+    "actor": {
+      "entity": {
+        "browserSettings": {
+          "browserMonitoring": {
+            "loader": "SPA",
+            "pinnedVersion": null
+          }
+        },
+        "guid": "YOUR_BROWSER_APP_GUID",
+        "name": "Example Single Page App"
+      }
+    }
+  }
+}
+```
+
+In this example, the `loader` is `SPA` type and the `pinnedVersion` is `null`, which indicates that there is no version pinned.
+
+You may run this query at any time to verify your combination of loader and pinned version.
+
+### Step 2. Retrieve the Available Versions
+
+Find the desired agent version from the [New Relic browser agent releases page.](https://docs.newrelic.com/docs/browser/browser-monitoring/getting-started/browser-agent-eol-policy/)
+
+The `Browser agent release` column contains all of the currently supported browser agent versions. Make note of the semantic version number you want to pin.
+
+### Step 3. Pin the Desired Version
+
+Once you've identified the version you want to pin, use the following GraphQL mutation to set it.
+
+*Hint:* Specify the semantic version of the browser agent that you would like your app to use. Leave this blank to use the most recent version. Use `'x'` in place of a numeric digit to represent the latest release within the version range. For example, `'1.x.x'`
+
+```graphql
+mutation  {
+  agentApplicationSettingsUpdate(
+    guid: "YOUR_BROWSER_APP_GUID"
+    settings: {browserMonitoring: {pinnedVersion: "1.229.0"}}
+  ) {
+    browserProperties {
+      jsLoaderScript
+    }
+    browserSettings {
+      browserMonitoring {
+        loader
+        pinnedVersion
+      }
+    }
+  }
+}
+
+```
+
+Note that the return fields for this mutation will contain the updated `pinnedVersion` value, as well as the updated `jsLoaderScript`.
+
+Changing the `loader` value to `NONE` will remove any pinned semantic version. 
+
+### Step 4. Decode the JavaScript Snippet (if using Copy and Paste)
```

Comment:
Maybe step 4 should be "Insert snippet into you application"? In the details, you could tell users that if they used the copy/paste routine, they'll need to decode the JavaScript snippet into HTML before they insert it. 

Still, I'm not really clear about how this works. The mutation generates JavaScript, so doesn't the JavaScript remain the same, but they just insert it between <script> tags?

Reply:
> Maybe step 4 should be "Insert snippet into you application"?

I'm okay with this change.

> The mutation generates JavaScript, so doesn't the JavaScript remain the same, but they just insert it between <script> tags?

The JavaScript snippet would change if the customer changes the pinned version.

Reply:
Maybe, step 4 could just be combined into Step 5

Reply:
Oh I understand what you are asking.

`jsLoaderScript` is actually HTML `<script>` tag that contains the JavaScript agent.

An example response:

```json
{
  "data": {
    "actor": {
      "entity": {
        "browserProperties": {
          "jsLoaderScript": "<script type=\"text/javascript\">\n;window.NREUM|| ... p\n</script>\n"
....
```

> Would users just copy the jsLoaderScriptJavaScript from the mutation and insert it into their existing <script> section of their apps? 

No, users would replace their existing HTML script element.

> What's involved in "decoding"? 

The GraphQL mutation result returns a JSON-encoded string. I.e., quotation marks are escaped and there are newline literals....:

```
<script type=\"text/javascript\">\n;window.NREUM|| ... p\n</script>\n"
```

So to have valid JavaScript, a user would have to JSON-decode it to escape the slashes and interpret the newlines.

In my prior experience, I paste the entire string into a tool called "Boop" and use the "JavaScript Eval" functionality. It runs the input as JavaScript code, and the output is text that is valid `<script>` and JavaScript inside.

Theoretically, a developer could use a different method to do it, as long as the main constraints are 1. removing slash mark escaping from quotation marks and 2. removing newline literals.

> Is it more than just nesting the JavaScript inside the <script> tags?

They will not need to nest JavaScript code in HTML <script> tags because the API provides both of those.

> Also, where would they put it in their existing <script> code? 

They would remove their existing script and replace it with this one. The point of the version pinning API is so that they can "freeze" their browser agent from getting updates. Also if they keep their existing script and add this one, I'm not sure how that would work. "It's not supported," I suppose!

> At the end?

This script would be pasted into the <head> tags, where the old one went. By following the directions in the copy/paste documentation they will be ok. 👍 

> If users are not using the copy/paste routine, what else would they do?

For users who enable browser monitoring through an APM agent like Python or Ruby, they would have to restart their agent for the new changes to take effect. From then on, any new web pages served will be automatically instrumented with their desired browser agent with the pinned version they specified.

Reply:
Thank you for the explanation, @ajtran303!